deployments/helm/KubeArmorOperator/templates/helpers.tpl

```diff
@@ -19,4 +19,39 @@
 {{- else }}
 {{- printf "%s:%s" .Values.kubearmorOperator.image.repository .Values.kubearmorOperator.image.tag }}
 {{- end }}
+{{- end }}
+
+{{- define "generateArgs" }}
+{{- $result := list -}}
+{{- $annotArgExists := false -}}
+{{- $initDeployArgExists := false -}}
+{{- range .Values.kubearmorOperator.args -}}
+  {{- if hasPrefix "--annotateResource" . -}}
+    {{- $annotArgExists = true -}}
+  {{- end -}}
+  {{- if hasPrefix "--initDeploy" . -}}
+    {{- $initDeployArgExists = true -}}
+  {{- end -}}
+  {{- $result = append $result . -}}
+{{- end -}}
+{{- $tag := (split ":" (include "operatorImage" .))._1 -}}
+{{- if (not $annotArgExists) -}}
+  {{- if or (eq $tag "latest") (and (hasPrefix "v" $tag) (semverCompare "^1.5.0" $tag)) }}
+    {{- if .Values.kubearmorOperator.annotateResource -}}
+      {{- $result = append $result "--annotateResource=true" -}}
+    {{- else -}}
+      {{- $result = append $result "--annotateResource=false" -}}  
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
+{{- if (not $initDeployArgExists) -}}
+  {{- if or (eq $tag "latest") (and (hasPrefix "v" $tag) (semverCompare "^1.4.0" $tag)) }}
+    {{- if .Values.kubearmorOperator.initDeploy -}}
+      {{- $result = append $result "--initDeploy=true" -}}
+    {{- else -}}
+      {{- $result = append $result "--initDeploy=false" -}}
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
+{{ toYaml $result }}
```

Comment:
Can you explain this 😅

Reply:
Sure, but I'm thinking of completely removing arg handling this way though. Let's wait for changes I'm planning to push then we'll discuss.